dsm/resource_group.go

```diff
@@ -123,36 +231,38 @@ func resourceReadGroup(ctx context.Context, d *schema.ResourceData, m interface{
 				return diag.FromErr(err)
 			}
 		}
+		if _, ok := req["hmg"]; ok {
+			if err := d.Set("hmg", fmt.Sprintf("%v", req["hmg"])); err != nil {
+				return diag.FromErr(err)
+			}
+		}
 	}
 	return diags
 }
 
-// [U]: Update Group
-func resourceUpdateGroup(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	return nil
-}
-
 // [D]: Delete Group
 func resourceDeleteGroup(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 
-	_, statuscode, err := m.(*api_client).APICall("DELETE", fmt.Sprintf("sys/v1/groups/%s", d.Id()))
-	if (err != nil) && (statuscode != 404) && (statuscode != 400) {
+	dataSourceGroupRead(ctx, d, m)
+	group_id := d.Get("group_id").(string)
+
+	_, statuscode, err := m.(*api_client).APICall("DELETE", fmt.Sprintf("sys/v1/groups/%s", group_id))
+	if err != nil {
 		diags = append(diags, diag.Diagnostic{
 			Severity: diag.Error,
 			Summary:  "[DSM SDK] Unable to call DSM provider API client",
-			Detail:   fmt.Sprintf("[E]: API: DELETE sys/v1/groups: %v", err),
+			Detail:   fmt.Sprintf("[E]: API: DELETE sys/v1/groups/%s %d: %v", group_id, statuscode, err),
+		})
+		return diags
+	}
+	if statuscode == 400 {
```

Comment:
Why 400 specifically instead of `!= 200`?

Reply:
Well, I don't know. It has been set like that way back. We will be circling back to these issues in the next release.